packages/algob/src/internal/deployer.ts

```diff
@@ -147,6 +151,36 @@ class DeployerBasicMode {
     }
   }
 
+  /**
+   * Returns SSC checkpoint key using application index,
+   * returns undefined if it doesn't exist
+   * @param index Application index
+   */
+  getSSCCPKeyFromId (index: number): string | undefined {
+    const resultMap = this.cpData.precedingCP[this.networkName]?.ssc ?? new Map();
+    for (const [key, nestedMap] of resultMap) {
+      if ([...nestedMap][nestedMap.size - 1][1].appID === index) {
+        return key;
+      }
+    }
+    return undefined;
+  }
+
+  /**
+   * Returns ASA checkpoint key using asset index,
+   * returns undefined if it doesn't exist
+   * @param index Asset Index
+   */
+  getASAKeyFromId (index: number): string | undefined {
+    const resultMap = this.cpData.precedingCP[this.networkName]?.asa ?? new Map();
+    for (const [key, value] of resultMap) {
+      if (value.assetIndex === index) {
+        return key;
+      }
+    }
+    return undefined;
+  }
```

Comment:
would be good to have a namespace for checkpoint related functions, then it will be easier to refer this methods.

Reply:
i was thinking to put them in checkpoint interface, but because checkpoints are only accessible to deployer,
it couldn't be used in `executeTransaction`. did you mean this ? or something else

Reply:
I was thinking about having a checkpoint operator and use it in deployer, so this way we would need to do:
```
deployer.checkpoint.appKeyFromIndex(...)  // here checkpoint is a new class
// other methods would need to be modified as well.
```
What do you think?

Reply:
yes, this would be good